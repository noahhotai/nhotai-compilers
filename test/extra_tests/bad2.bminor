// Array init type mismatch

a: array [2] array [2] integer = {{1, 2}, {3, 4}};
b: array [2] array [2] array [1] float = {{{1}, {2}}, {{3}, {4}}};

// wrong type
c: array [2] array [2] integer = {{1, 2}, {3.2, "hello"}};

// too many nested arrays
z: array [2] array [2] integer = {{{1}, {2}}, {{3}, {4}}};

// not enough arrays
t: array [2] array [2] integer = {1, 2};

main: function integer () = {
    // Assigning the wrong value to array type
    a[1][0] = 3.5;
    b[0][0][0] = "hello";
}
