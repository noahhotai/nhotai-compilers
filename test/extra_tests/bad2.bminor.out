Program parsed successfully.
a resolves to global a
b resolves to global b
Program resolved successfully.
type error: cannot assign  ({{{1},{2}},{{3},{4}}}) to type array [2] array [2] array [1] float (b).
type error: cannot assign  ({{1, 2},{3.200000, "hello"}}) to type array [2] array [2] integer (c).
type error: cannot assign  ({{{1},{2}},{{3},{4}}}) to type array [2] array [2] integer (z).
type error: cannot assign  ({1, 2}) to type array [2] array [2] integer (t).
type error: cannot assign type float (3.500000) to type float (a[1][0]).
type error: cannot assign type string ("hello") to type string (b[0][0][0]).
Program failed to typecheck.
